target/linux/generic/files/drivers/mtd/mtdsplit/mtdsplit_uimage.c

```diff
@@ -321,6 +321,55 @@ static struct mtd_part_parser uimage_netgear_parser = {
 	.type = MTD_PARSER_TYPE_FIRMWARE,
 };
 
+
+/**************************************************
+ * ALLNET
+ **************************************************/
+
+#define FW_MAGIC_SG8208M	0x00000006
+#define FW_MAGIC_SG8310PM	0x83000006
+
+static ssize_t uimage_verify_allnet(u_char *buf, size_t len, int *extralen)
+{
+	struct uimage_header *header = (struct uimage_header *)buf;
+
+	switch (be32_to_cpu(header->ih_magic)) {
+	case FW_MAGIC_SG8208M:
+	case FW_MAGIC_SG8310PM:
+		break;
+	default:
+		return -EINVAL;
+	}
+
+	if (header->ih_os != IH_OS_LINUX)
+		return -EINVAL;
+
+	return 0;
+}
+
+static int
+mtdsplit_uimage_parse_allnet(struct mtd_info *master,
+			      const struct mtd_partition **pparts,
+			      struct mtd_part_parser_data *data)
+{
+	return __mtdsplit_parse_uimage(master, pparts, data,
+				      uimage_verify_allnet);
+}
+
+static const struct of_device_id mtdsplit_uimage_allnet_of_match_table[] = {
+	{ .compatible = "allnet,uimage" },
+	{},
+};
+
+static struct mtd_part_parser uimage_allnet_parser = {
+	.owner = THIS_MODULE,
+	.name = "allnet-fw",
+	.of_match_table = mtdsplit_uimage_allnet_of_match_table,
+	.parse_fn = mtdsplit_uimage_parse_allnet,
+	.type = MTD_PARSER_TYPE_FIRMWARE,
```

Comment:
Can you drop `.type` assignment (and verify it works)? That is required for pre-DT devices only that don't specify `firmware` partition type directly. As long as you use `allnet,uimage` in DT there should be no need for the `MTD_PARSER_TYPE_FIRMWARE`.

Reply:
The .type can be dropped. I tested it, the partition is still split.